python/lsst/ap/association/trailedSourceFilter.py

```diff
@@ -0,0 +1,110 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""A simple implementation of source association task for ap_verify.
+"""
+
+__all__ = ["TrailedSourceFilterTask", "TrailedSourceFilterConfig"]
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from lsst.utils.timer import timeMethod
+
+# import numpy as np
+import pandas as pd
+
+# Enforce an error for unsafe column/array value setting in pandas.
+pd.options.mode.chained_assignment = 'raise'
+
+
+class TrailedSourceFilterConfig(pexConfig.Config):
+    """Config class for TrailedSourceFilterTask.
+    """
+    maxTrailLength = pexConfig.Field(
+        dtype=float,
+        doc='Maximum trail length permitted is less than 10 degrees/day. This is a rate '
+            'of 0.416 arcseconds per second.As trail length is measured in '
+            'arcseconds, it is dependant on the length of the exposure.',
+        default=0.416,  # HSC Default. Should Decam and LSST defaults be passed? Does it change possibly?
+    )
+
+
+class TrailedSourceFilterTask(pipeBase.Task):
+    """Find trailed sources in DIAObjects.
+
+    """
+
+    ConfigClass = TrailedSourceFilterConfig
+    _DefaultName = "trailedAssociation"
+
+    @timeMethod
+    def run(self,
+            dia_sources, exposure):
+        """Find trailed sources which have not been filtered out and will
+        not be included in the diaSource catalog.
+
+        Parameters
+        ----------
+        dia_sources : `pandas.DataFrame`
+            New DIASources to be checked for trailed sources.
+        exposure : `pandas.DataFrame`
+            Calibrated exposure differenced with a template image during
+            image differencing.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            Results struct with components.
+
+            - ``"dia_sources"`` : DiaSource table that is free from unwanted
+             trailed sources (`pandas.DataFrame`)
+
+            - ``"trailed_dia_sources"`` : DiaSources that have trailed more than
+             0.416 arcseconds/second*exposure_time(`pandas.DataFrame`)
+        """
+        trail_mask = self.check_dia_source_trail(dia_sources, exposure)
+
+        return pipeBase.Struct(
+            diaSources=dia_sources[~trail_mask].reset_index(drop=True),
+            trailedDiaSources=dia_sources[trail_mask].reset_index(drop=True))
+
+    def check_dia_source_trail(self, dia_sources, exposure):
```

Comment:
This should be private (prefix with `_`): we didn't do that with older tasks, and have come to regret it. The only public entry point for tasks should be `run`.